src/app/data-model/Decode.h

```diff
@@ -103,6 +103,27 @@ CHIP_ERROR Decode(TLV::TLVReader & reader, X & x)
     return x.Decode(reader);
 }
 
+/*
+ * @brief
+ *
+ * This specific variant that decodes cluster objects (like structs, commands, events) from TLV
+ * depends on the presence of a Decode method on the object to present. The signature of that method
+ * is as follows:
+ *
+ * CHIP_ERROR <Object>::Decode(TLVReader &reader);
+ *
+ */
+template <typename X,
+          typename std::enable_if_t<std::is_class<X>::value &&
+                                        std::is_same<decltype(std::declval<X>().Decode(std::declval<TLV::TLVReader &>(),
+                                                                                       std::declval<Optional<FabricIndex> &>())),
+                                                     CHIP_ERROR>::value,
+                                    X> * = nullptr>
+CHIP_ERROR Decode(TLV::TLVReader & reader, X & x, const Optional<FabricIndex> & overwriteFabricIndex)
```

Comment:
Is it "overwrite" as here or "override" as in other places in this PR?

And more to the point: what are the semantics of that Optional argument?  What does it actually do?  Needs to be documented.

Reply:
But why is it optional though?